sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala

```diff
@@ -78,4 +78,30 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
       }}
     }
   }
+
+  test("Read row group containing both dictionary and plain encoded pages") {
+    spark.conf.set("parquet.dictionary.page.size", "2048")
+    spark.conf.set("parquet.page.size", "4096")
```

Comment:
Let's use `withSQLConf` to alter these settings so that they are automatically reverted to their original values at the end of the scope.